### DISCUSSION TITLE
How to set cookies from tRPC to client?

### DISCUSSION THREAD

**Discussion by Extarys** [trpc/trpc#3769] [Q&A] [answered]
All examples of this are from 2021 or starts of 2022  and no longer works, or is about how to `get` cookies. I guess this is the next "big" thing once setting cookies work. It seems like not many people use cookies :joy: I'm about to pull the hairs out. :exploding_head: 
I'm currently trying the trpc-nuxt plugin, but so far I move slow and break everything. Your wisdom is tremendously appreciated.

**Comment by is-it-ayush** [comment]
Normally, I return the cookie value to the client & handle the store logic there. You can use packages to do this for you, there's so many libraries out there. Here's a link to one of my older projects that approach this method I described. https://github.com/is-it-ayush/athing/blob/77ea90dcc7e8bcabec9e517a0958eb7e2393f710/src/pages/auth/login/index.tsx#L52. I've used a package called [ncookies](https://www.npmjs.com/package/nookies) there! 🙂 
I'm not sure this is the best method, but it should work.

**Reply by Extarys** [reply]
Thanks for your reply. I should've included code I know, I was just so angry at the whole thing I turned my computer off right after that post :joy: Got it working this morning.

**Reply by is-it-ayush** [reply]
@Extarys lmao. Good luck with your project!

**Comment by Extarys** [comment, marked as answer]
Got it working, it seems.

Context
```
export const createContext = (event: H3Event) => {
	if (!prisma) {
		prisma = new PrismaClient({
			log: env.NODE_ENV === 'development' ? ['query', 'error', 'warn'] : ['error'],
		})
	}
	return {
		ip: event.node.req.headers['x-forwarded-for'] || event.node.req.socket.remoteAddress || event.node.req.headers.host,
		node: event.node, // Pass the whole H3 Event
		prisma
	}
}
```

`router.query()`

```
setCookie(event.ctx, 'TRPC', 'DATAAA'); // setCookie is looking for 'node'
```
